src/frameworks/next/index.ts

```diff
@@ -84,77 +96,104 @@ export async function build(dir: string): Promise<BuildResult> {
     throw e;
   });
 
-  try {
-    // Using spawn here, rather than their programatic API because I can't silence it
-    // Failures with Next export are expected, we're just trying to do it if we can
-    execSync(`${CLI_COMMAND} export`, { cwd: dir, stdio: "ignore" });
-  } catch (e) {
-    // continue, failure is expected
+  const reasonsForBackend = [];
+  const { distDir } = await getConfig(dir);
+
+  const middlewareManifest = await readJSON<MiddlewareManifest>(
+    join(dir, distDir, "server", MIDDLEWARE_MANIFEST)
+  );
+  const usingMiddleware = isUsingMiddleware(middlewareManifest.middleware);
+  if (usingMiddleware) {
+    reasonsForBackend.push("middleware");
   }
 
-  let wantsBackend = true;
-  const { distDir } = await getConfig(dir);
-  const exportDetailPath = join(dir, distDir, "export-detail.json");
-  const exportDetailExists = await pathExists(exportDetailPath);
-  const exportDetailBuffer = exportDetailExists ? await readFile(exportDetailPath) : undefined;
-  const exportDetailJson = exportDetailBuffer && JSON.parse(exportDetailBuffer.toString());
-  if (exportDetailJson?.success) {
-    const appPathRoutesManifestPath = join(dir, distDir, "app-path-routes-manifest.json");
-    const appPathRoutesManifestJSON = fileExistsSync(appPathRoutesManifestPath)
-      ? await readFile(appPathRoutesManifestPath).then((it) => JSON.parse(it.toString()))
-      : {};
-    const prerenderManifestJSON = await readFile(
-      join(dir, distDir, "prerender-manifest.json")
-    ).then((it) => JSON.parse(it.toString()));
-    const anyDynamicRouteFallbacks = !!Object.values(
-      prerenderManifestJSON.dynamicRoutes || {}
-    ).find((it: any) => it.fallback !== false);
-    const pagesManifestJSON = await readFile(
-      join(dir, distDir, "server", "pages-manifest.json")
-    ).then((it) => JSON.parse(it.toString()));
-    const prerenderedRoutes = Object.keys(prerenderManifestJSON.routes);
-    const dynamicRoutes = Object.keys(prerenderManifestJSON.dynamicRoutes);
-    const unrenderedPages = [
-      ...Object.keys(pagesManifestJSON),
-      // TODO flush out fully rendered detection with a app directory (Next 13)
-      // we shouldn't go too crazy here yet, as this is currently an expiriment
-      ...Object.values<string>(appPathRoutesManifestJSON),
-    ].filter(
-      (it) =>
-        !(
-          ["/_app", "/", "/_error", "/_document", "/404"].includes(it) ||
-          prerenderedRoutes.includes(it) ||
-          dynamicRoutes.includes(it)
-        )
-    );
-    // TODO log these as a reason why Cloud Functions are needed
-    if (!anyDynamicRouteFallbacks && unrenderedPages.length === 0) {
-      wantsBackend = false;
+  const { isNextImageImported } = await readJSON<ExportMarker>(join(dir, distDir, EXPORT_MARKER));
```

Comment:
I know outside of the scope here but maybe we could refactor into an usingImageOptimization util fn?

Reply:
done! am planning to do the same for almost everything that affects `reasonsForBackend`, e.g. `isUsingAppDirectory`, `isUsingFallback`, `isUsingRevalidate`, ...